client/src/app-context.tsx

```diff
@@ -1,33 +1,67 @@
 import { useState, createContext, ReactNode, FC, useMemo, useCallback } from 'react'
 
+// Keep the new interfaces from server_timothy branch
+export interface UploadedImage {
+    file: File;
+    preview: string;
+    serverResponse?: UploadResponse;
+}
+
+export interface UploadResponse {
+    status: string;
+    message: string;
+    details: {
+        original_size: number;
+        filename: string;
+        mock_upscale_factor: number;
+    };
+}
+
+// Combine both AppState interfaces
 export interface AppState {
-    imageData: ImageData | null,
-    updateAppState: (newState: Partial<AppState>) => void
+    image: UploadedImage | null;
+    imageData: ImageData | null;
```

Comment:
We don't need to return image field at all. Take a look at EditorContext: https://github.com/Tucchhaa/image-filter-editor/blob/main/client/src/editor/editor-context.tsx#L23, it has ImageHistory object.

ImageHistory object contains last N generated images (filtered or upscaled). So if we want to filter image or upscale it, we apply it to the last image. That will be useful in this scenario: 
1. A user uploaded an image -> 2. Applied some filter -> 3. Upscaled it -> 4. Applied another filter -> 5. Upscaled it again.

So we need to send to the server last ImageData object in the ImageHistory. To do that you first need to convert ImageData to blob and then send it (or find another way to send ImageData).

Reply:
Just a bit of info why we need ImageHistory object: it will be helpful for reverting to previous images (aka ctrl+z)